src/main/java/org/terasology/cities/bldg/gen/CommercialBuildingGenerator.java

```diff
@@ -53,15 +51,15 @@ public Building generate(Parcel parcel, HeightMap hm) {
         Orientation o = parcel.getOrientation();
         DefaultBuilding bldg = new DefaultBuilding(o);
 
-        Rect2i rc = parcel.getShape().expand(-4, -4);
-        Rect2i roofRc = rc.expand(2, 2);
+        BlockArea rc = parcel.getShape().expand(-4, -4, new BlockArea(BlockArea.INVALID));
+        BlockArea roofRc = rc.expand(2, 2);
```

Comment:
This will now modify the rectangle `rc` in place, leading to errors in the generated buildings (likely the same for the other generators :see_no_evil: ). 

Previously, `Rect2i#expand` did create a new instance, so this was not an issue.